bftengine/src/preprocessor/PreProcessor.hpp

```diff
@@ -161,7 +161,20 @@ class PreProcessor {
   void handleSingleClientRequestMessage(ClientPreProcessReqMsgUniquePtr clientMsg,
                                         bool arrivedInBatch,
                                         uint16_t msgOffsetInBatch);
-
+  bool isRequestPreProcessingRightNow(const RequestStateSharedPtr &reqEntry,
+                                      ReqId reqSeqNum,
+                                      NodeIdType clientId,
+                                      NodeIdType senderId);
+  bool isRequestAlreadyExecuted(ReqId reqSeqNum, NodeIdType clientId, const std::string &cid);
+  bool isRequestPreProcessedBefore(const RequestStateSharedPtr &reqEntry,
+                                   SeqNum reqSeqNum,
+                                   NodeIdType clientId,
+                                   const std::string &cid);
+
+  bool isRequestPassingConsensusOrPostExec(SeqNum reqSeqNum,
```

Comment:
I see you extracted those functions from `handleSingleClientRequestMessage()`. Are there any changes to the logic to look out for, or is it a pure refactoring?

Reply:
Pure refactoring

Reply:
Hi Yulia, thanks for the clarification - looks great!

Reply:
@teoparvanov Thank you for the review!